src/charts/CytoViz/CytoViz.js

```diff
@@ -101,7 +101,36 @@ export const CytoViz = (props) => {
       const selectedElement = e.target;
       setCurrentElementDetails(getElementDetailsCallback(selectedElement));
     });
+    cytoscapeRef.on('click', 'node', function (e) {
+      cytoscapeRef
+        .elements()
+        .not(e.target)
+        .forEach((nodeOrEdge) => {
+          nodeOrEdge.unselect();
+          if (nodeOrEdge.isEdge()) {
+            nodeOrEdge.data('asInEdgeHighlighted', false);
+            nodeOrEdge.data('asOutEdgeHighlighted', false);
+          }
+        });
+
+      const selectedElement = e.target;
+      selectedElement.select();
+      selectedElement.outgoers('edge').forEach((edge) => edge.data('asOutEdgeHighlighted', true));
+      selectedElement.incomers('edge').forEach((edge) => edge.data('asInEdgeHighlighted', true));
+
+      setCurrentElementDetails(getElementDetailsCallback(selectedElement));
+    });
     cytoscapeRef.on('unselect', 'node, edge', function (e) {
+      cytoscapeRef
+        .elements()
+        .not(e.target)
+        .forEach((nodeOrEdge) => {
+          nodeOrEdge.unselect();
+          if (nodeOrEdge.isEdge()) {
+            nodeOrEdge.data('asInEdgeHighlighted', false);
+            nodeOrEdge.data('asOutEdgeHighlighted', false);
+          }
+        });
```

Comment:
As discussed, you can move your code from the `click` listener to the `select` listener to support the multi-selection feature too (_shift + mouse selection_).

I think we don't need the `nodeOrEdge.unselect()` in the loop anymore, so there are several improvements that are now possible :
- the `forEach` loops to set data can be replaced by the `eles.data` function (doc [here](https://js.cytoscape.org/#eles.data))
- the `nodeOrEdge.isEdge()` check in the loop can be replaced by the initial selection `cytoscapeRef.edges()`
- checking for "not the target" can be removed if we set the value for the selected element afterwards :wink: 


Here's a suggestion for the `click` and `unselectect` listeners:
```suggestion
      const selectedElement = e.target;
      selectedElement.select(); // Make sure the last element appears as selected even during multi-selection
      setCurrentElementDetails(getElementDetailsCallback(selectedElement));
      // Reset highlighted edges for all edges, then set the data only for in/out edges of the selected node
      cytoscapeRef.edges().data({ asInEdgeHighlighted: false, asOutEdgeHighlighted: false });
      selectedElement.outgoers('edge').data('asOutEdgeHighlighted', true);
      selectedElement.incomers('edge').data('asInEdgeHighlighted', true);
    });

    cytoscapeRef.on('unselect', 'node, edge', function (e) {
      cytoscapeRef.edges().data({ asInEdgeHighlighted: false, asOutEdgeHighlighted: false });
```